### PR TITLE
Verificar la creación de un nuevo Personal

### PR DESCRIPTION
**Pull request by franzelunico**
image
image

### CODE REVIEW

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    private WebElement selectAreaPersonnel;
+
+    @FindBy(xpath = "//span[@class='mat-button-wrapper'][contains(text(),'Enviar')]")
+    private WebElement buttonSendPersonnel;
+
+    public AddPersonnel() {
+        CommonEvents.isVisible(inputNamePersonnel);
+    }
+
+    public void setName(String name) {
```
Comment (ivanMise) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java]: maybe you could change to: setInputFieldName(String name)

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        CommonEvents.setInputField(inputNamePersonnel, name);
+    }
+
+    public void setLastName(String lastName) {
```
Comment (ivanMise) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java]: maybe you could change to: setInputFieldLastName(String lastName)

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        CommonEvents.setInputField(inputLastNamePersonnel, lastName);
+    }
+
+    public void setEmail(String email) {
```
Comment (ivanMise) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java]: maybe you could change to: setInputFieldEmail(String email)

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        CommonEvents.setInputField(inputEmailPersonnel, email);
+    }
+
+    public void setAddress(String address) {
```
Comment (ivanMise) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java]: maybe you could change to: setInputFieldAddress(String address)

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        CommonEvents.setInputField(inputAddressPersonnel, address);
+    }
+
+    public void setPhone(String phone) {
```
Comment (ivanMise) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java]: maybe you could change to: setInputFieldPhone(String phone)

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        CommonEvents.setInputField(inputPhonePersonnel, phone);
+    }
+
+    public void setArea() {
+            CommonEvents.clickButton(optionAreaPersonnel);
```
Comment (ivanMise) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java]: maybe you could change to: setOptionArea()

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+            CommonEvents.clickButton(selectAreaPersonnel);
+    }
+
+    public boolean send() {
```
Comment (ivanMise) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/AddPersonnel.java]: complement the name of this methodo send () or maybe you could change to: clickOnButtonSendPersonnel ()

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ButtonAddPersonnel.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+public class ButtonAddPersonnel extends BasePage {
+
+    @FindBy(xpath = "//span[@class='mat-button-wrapper'][contains(text(),'Agregar nuevo personal')]")
+    private WebElement buttonAddPersonnel;
+
+    public ButtonAddPersonnel(){
+        CommonEvents.isVisible(buttonAddPersonnel);
+    }
+
+    public boolean validateButtonAddPersonnel(){
```
Comment (ivanMise) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ButtonAddPersonnel.java]: maybe you could change to: validateButtonAddPersonnelIsVisible()

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ButtonAddPersonnel.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+        return CommonEvents.isVisible(buttonAddPersonnel);
+    }
+
+    public AddPersonnel clickAgregarPersonal(){
```
Comment (ivanMise) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personnel/ButtonAddPersonnel.java]: use english please.
maybe you could change to: clickOnButtonAddPersonnel()

**Review by ivanMise — APPROVED** [review]

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 10]
```diff
@@ -7,6 +7,7 @@
 import org.umssdiplo.automationv01.core.managepage.Personnel.ButtonAddPersonnel;
 import org.umssdiplo.automationv01.core.managepage.Personnel.PersonnelSearch;
 import org.umssdiplo.automationv01.core.managepage.Usuario.ListUser;
+import org.umssdiplo.automationv01.core.managepage.Personnel.ButtonAddPersonnel;
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java:10]: review the pageobject buttonaddpersonnel, maybe this button should be in AddPersonnel

File: src/test/resources/features/Personnel/addPersonnel.feature [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario Outline: Verificar la creacion de un nuevo Personal.
+    Given 'Menu principal' estee cargado
+    And seleccionar menu 'Personal' en la pagina 'Menu Principal'
+    And click en el submenu 'Personal'
+    Given click en el boton 'Agregar nuevo personal'
```
Comment (jdpapiero) [inline src/test/resources/features/Personnel/addPersonnel.feature:12]: already exist a Given in this feature

File: src/test/resources/features/Personnel/addPersonnel.feature [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+    When llenar el campo nombre <nombre>
+    When llenar el campo apellido <apellido>
```
Comment (jdpapiero) [inline src/test/resources/features/Personnel/addPersonnel.feature:14]: Already exist 'When' consecutive in this feature, maybe the correct sintaxt should be And.
When
And
And
And
Then

File: src/test/resources/features/Personnel/addPersonnel.feature [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+    When llenar el campo email <email>
```
Comment (jdpapiero) [inline src/test/resources/features/Personnel/addPersonnel.feature:15]: Already exist 'When' consecutive in this feature, maybe the correct sintaxt should be And.
When
And
And
And
Then

File: src/test/resources/features/Personnel/addPersonnel.feature [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+    When llenar el campo direccion <direccion>
```
Comment (jdpapiero) [inline src/test/resources/features/Personnel/addPersonnel.feature:16]: Already exist 'When' consecutive in this feature, maybe the correct sintaxt should be And.
When
And
And
And
Then

File: src/test/resources/features/Personnel/addPersonnel.feature [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+    When llenar el campo telefono <telefono>
```
Comment (jdpapiero) [inline src/test/resources/features/Personnel/addPersonnel.feature:17]: Already exist 'When' consecutive in this feature, maybe the correct sintaxt should be And.
When
And
And
And
Then

File: src/test/resources/features/Personnel/addPersonnel.feature [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+    When seleccionar el 'Area' de instalaciones hidricas
```
Comment (jdpapiero) [inline src/test/resources/features/Personnel/addPersonnel.feature:18]: Already exist 'When' consecutive in this feature, maybe the correct sintaxt should be And.
When
And
And
And
Then

**Review by jdpapiero — CHANGES_REQUESTED** [review]
please review observations, this PR would by closed until fix them

**Comment by jdpapiero** [comment]
please review observations, this PR would by closed until fix them

**Pull request closed without merging**